runners/runner-local/src/manager.rs

```diff
@@ -0,0 +1,215 @@
+            .ok_or(LocalRunnerError::Internal)
+            .attach_printable("Can't pick a port for the status server")?;
+
+        let status_server_address = format!("0.0.0.0:{port}");
+
+        let status_server_args = &[
+            "--status-server-address".to_string(),
+            status_server_address.clone(),
+        ];
+
+        let status_server_address = format!("http://{status_server_address}");
+
+        let cmd = format!(
+            "{} {} {}",
+            command.program,
+            command.args.join(" "),
+            status_server_args.join(" ")
+        );
+        info!("Starting indexer {} with command `{}`", indexer.name, cmd);
+
+        let logs_dir = "logs";
+
+        let (stdout, stderr) = create_dir_all(logs_dir)
+            .change_context(LocalRunnerError::Internal)
+            .attach_printable("cannot create logs folder")
+            .and_then(|_| {
+                let stdout_path = PathBuf::from(format!("{logs_dir}/{}-stdout.log", indexer.name));
+                let stderr_path = PathBuf::from(format!("{logs_dir}/{}-stderr.log", indexer.name));
+
+                let stdout_file = File::create(&stdout_path)
+                    .change_context(LocalRunnerError::Internal)
+                    .attach_printable(format!(
+                        "cannot create stdout file `{}`",
+                        stdout_path.display()
+                    ))?;
+
+                let stderr_file = File::create(&stderr_path)
+                    .change_context(LocalRunnerError::Internal)
+                    .attach_printable(format!(
+                        "cannot create stderr file `{}`",
+                        stderr_path.display()
+                    ))?;
+
+                let stdout = Stdio::from(stdout_file);
+                let stderr = Stdio::from(stderr_file);
+
+                info!(
+                    "Writing indexer {} stdout to `{}`",
+                    indexer.name,
+                    &stdout_path.display()
+                );
+                info!(
+                    "Writing indexer {} stderr to `{}`",
+                    indexer.name,
+                    &stderr_path.display()
+                );
+
+                Ok((stdout, stderr))
+            })
+            .unwrap_or_else(|err| {
+                warn!(err =? err, "failed to write indexer logs in directory `{}`", logs_dir);
+                (Stdio::piped(), Stdio::piped())
+            });
+
+        let child = Command::new(command.program)
+            .current_dir(command.current_dir)
+            .envs(command.envs)
+            .args(command.args)
+            .args(status_server_args)
+            .stdout(stdout)
+            .stderr(stderr)
+            .spawn()
+            .change_context(LocalRunnerError::Internal)
+            .attach_printable("failed to spawn indexer")?;
+
+        // TODO: handle NotFound command which means the plugin is not installed separately from other errors
```

Comment:
Yep, this one was confusing for me because I didn't read the full PR comment!

Reply:
In the CLI we have some code to check for this. Maybe you can import that?

Reply:
We can't import it, but we can copy past it

Reply:
Yeah I think it's fine. Worth refactoring in another PR to share code between crates.